src/Bicep.Core/Configuration/ConfigHelper.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Configuration
+{
+    internal class ConfigHelper
+    {
+        private const string SettingsFileName = "bicepsettings.json";
+
+        /// <summary>
+        /// Property exposes the configuration root
+        /// that is currently loaded
+        /// </summary>
+        public IConfigurationRoot Config { get; private set; }
+
+        internal ConfigHelper()
+        {
+            this.Config = BuildConfig(Directory.GetCurrentDirectory());
+        }
+
+        private IConfigurationRoot BuildConfig(string localFolder)
+        {
+            var configBuilder = new ConfigurationBuilder();
+
+            // load the default settings from file embedded as resource
+            var assembly = Assembly.GetExecutingAssembly();
+            var names = assembly.GetManifestResourceNames();
+            var defaultConfigResourceName = names.FirstOrDefault(n => n.EndsWith(SettingsFileName));
+
+            // keep this stream open until after Build() call
+            using (var defaultConfigStream = assembly.GetManifestResourceStream(defaultConfigResourceName))
+            {
+                Debug.Assert(defaultConfigStream != null, "Default configuration file should exist as embedded resource.");
+                configBuilder.AddJsonStream(defaultConfigStream);
+
+                // last added json settings take precedent - add local settings last
+                if (DiscoverLocalConfigurationFile(localFolder) is string localConfig)
+                {
+                    configBuilder.AddJsonFile(localConfig, true, true);
+                    this.CustomSettingsFileName = localConfig;
+                }
+                else
+                {
+                    this.CustomSettingsFileName = default;
+                }
+
+                return configBuilder.Build();
+            }
+        }
+
+        private string? DiscoverLocalConfigurationFile(string? nextDir)
+        {
+            while (nextDir != default)
+            {
+                var fileName = Path.Combine(nextDir, SettingsFileName);
+                if (File.Exists(fileName))
+                {
+                    return fileName;
+                }
+                nextDir = Directory.GetParent(nextDir)?.FullName;
+            }
+            return default;
+        }
+
+        internal void LoadConfiguration(Uri fileUri)
+        {
+            var localFile = Path.GetDirectoryName(fileUri.LocalPath);
```

Comment:
You're assuming the URI is a local file path, which it might not be.  For instance, for an unsaved file, it might be "untitled:Untitled-1" (and then you end up with a fatal error).  Could be a URL file share as well, etc.

Use Uri.isFile to check.

Reply:
Must have for MVP.

Reply:
https://github.com/jfleisher/bicep/pull/14